### PR TITLE
Accomodate Timer API change

### PR DESCRIPTION
**Pull request by SimonDanisch**
Completely rewrote fpswhen. I'm not 100% sure, if it does exactly the same thing as the old fpswhen.
Also fixed type in timestamp and removed some anonymous functions.

### CODE REVIEW

File: src/timing.jl [hunk 1]
```diff
@@ -9,9 +9,8 @@ export every, fpswhen, fps, timestamp
 function every(delta::Float64)
     i = Input(time())
     update(timer) = push!(i, time())
-    t = Timer(update)
-    start_timer(t, delta, delta)
-    return lift(x->x, i, typ=Float64) # prevent push!
+    t = Timer(update, delta, delta)
+    return lift(identity,  i) # prevent push!
```
Comment (SimonDanisch) [inline src/timing.jl]: > prevent push!
Do we need the same mechanism for fpswhen?
I'm not sure how this prevents "push!"

File: src/timing.jl [hunk 2]
```diff
@@ -9,9 +9,8 @@ export every, fpswhen, fps, timestamp
 function every(delta::Float64)
     i = Input(time())
     update(timer) = push!(i, time())
-    t = Timer(update)
-    start_timer(t, delta, delta)
-    return lift(x->x, i, typ=Float64) # prevent push!
+    t = Timer(update, delta, delta)
```
Comment (sjkelly) [inline src/timing.jl]: I think this should be behind an `if VERSION >+ 0.4-<build>`.

File: src/timing.jl [hunk 3]
```diff
@@ -22,31 +21,20 @@ end
 #     freq: the maximum frequency at which fpswhen should update
 # Returns:
 #     an signal of Float64 time deltas
+function gate(wason_timer::Tuple{Bool, Timer}, ison::Bool, s::Input{Float64}, delta::Float64)
+    wason, timer = wason_timer
+    (!wason&&ison) && return (ison, Timer(x->push!(s, time()), 0, delta)) # start pushing again
```
Comment (sjkelly) [inline src/timing.jl]: Parens are not needed here

File: src/timing.jl (hunk 2 above)
Comment (sjkelly) [inline src/timing.jl]: The build is 5306

Comment (kmsquire) [inline src/timing.jl]: https://github.com/JuliaLang/Compat.jl/pull/103 should handle this once it's merged.

File: src/timing.jl (hunk 3 above)
Comment (SimonDanisch) [inline src/timing.jl]: true - I started doing this to make it clear which part is the condition and which is the statement to execute.
Some of my weakly enforced coding standards you could say...

File: src/timing.jl (hunk 2 above)
Comment (kmsquire) [inline src/timing.jl]: If you add a dependency to Compat 0.4.6 in `REQUIRE`, this should work on v0.3 as well.

**Pull request merged**